master/buildbot/steps/artifact.py

```diff
@@ -312,16 +312,19 @@ class CreateArtifactDirectory(ShellCommand):
     descriptionDone="Remote artifact directory created."
 
     def __init__(self,  artifactDirectory=None, artifactServer=None, artifactServerDir=None, artifactServerPort=None,
-                **kwargs):
+                 customArtifactPathRef=None, **kwargs):
         self.artifactDirectory = artifactDirectory
         self.artifactServer = artifactServer
         self.artifactServerDir = artifactServerDir
         self.artifactServerPort = artifactServerPort
+        self.customArtifactPathRef = customArtifactPathRef
         ShellCommand.__init__(self, **kwargs)
 
     def start(self):
         br = self.build.requests[0]
-        if mkdt(br.submittedAt) > ARTIFACT_LOCATION_CHANGE_DATE:
+        if callable(self.customArtifactPathRef):
```

Comment:
I think we should assume this a string, and not a callable (I might be wrong, but we'll find out)

Reply:
I've changed `customArtifactPathRef` to `customArtifactPath [str]`.